src/treebuilders/WaveletAdaptor.h

```diff
@@ -38,13 +38,58 @@ template <int D> class WaveletAdaptor : public TreeAdaptor<D> {
             , splitFac(sf) {}
     ~WaveletAdaptor() override = default;
 
+    void setPrecTree(std::vector<FunctionTree<D> *> treevec) { this->precTrees = treevec; }
+    void setMultiplicationSplit(bool multSplit) { this->multiplicationSplit = multSplit; };
+
 protected:
     bool absPrec;
     double prec;
     double splitFac;
+    std::vector<FunctionTree<D> *> precTrees;
+    bool multiplicationSplit = false;
 
     bool splitNode(const MWNode<D> &node) const override {
-        return node.splitCheck(this->prec, this->splitFac, this->absPrec);
+        if (multiplicationSplit and this->precTrees.size() > 0) {
+            auto multPrec = 1.0;
+            if (this->precTrees.size() != 2) std::cout << this->precTrees.size() << " ERROR " << std::endl;
```

Comment:
`MSG_ERROR("Invalid precTree size: " << this->precTrees.size());`

Reply:
Actually we should still discuss the details: I do not understand all the features that are "jumped over". 
What is "Splitfac" used for? 
In principle one could have more than two functions, but when is this used/useful?

Reply:
`splitFac` is explained in the [`crop()`](https://mrcpp.readthedocs.io/en/latest/mrcpp_api/mwfunctions.html#_CPPv4N5mrcpp12FunctionTree4cropEddb) function. I don't think we ever use anything else than `splitFac = 1.0`